website_sale_product_attribute_filter_visibility/tests/test_attribute_filter_visibility.py

```diff
@@ -9,16 +9,18 @@ def setUp(self):
         super(WebsiteSaleHttpCase, self).setUp()
         with self.cursor() as cr:
             env = self.env(cr)
+            self.all_attributes = env['product.attribute'].search([])
+            self.all_attributes.write({'website_published': False})
```

Comment:
Why unpublish all?

Reply:
For not change the test

Reply:
OK, but IMO, you should test both existence and not existence of one attribute, not only one side. Or are you doing that already (I have not followed full tour)?

Reply:
in the next lines I create two attributes, one published and other unpublished

Reply:
OK, got it! I saw the assignment line removed, but I was not counting with default value. Thanks for your patience.

Reply:
OK, forget my comments. It's Friday afternoon and I'm telling nonsense.